### ISSUE TITLE
chain.io not working because of firewall

### ISSUE THREAD

**Issue by mapodev** [Jacubeit/Bitcoin-MoneyMoney#6]
The API has a firewall set up that's why it's not working anymore.

I tried to change it to 

```function requestBitcoinQuantityForBitcoinAddress(bitcoinAddress)
  response = connection:request("GET", bitcoinRequestUrl(bitcoinAddress), {})
  json = JSON(response)
  balance = json:dictionary()['balance']
  return balance
end


-- Helper Functions
-- function convertSatoshiToBitcoin(satoshi)
--   return satoshi / 100000000
-- end

function cryptocompareRequestUrl()
  return "https://api.coinmarketcap.com/v1/ticker/bitcoin/?convert=EUR"
end

function bitcoinRequestUrl(bitcoinAddress)
  return "https://blockexplorer.com/api/addr/" .. bitcoinAddress .. "/"
end
```

But my extenstion is then not showed anymore. Someone help? Does it depend on the signature?

**Comment by derausdingsda** [comment]
yes, signature is the issue. you can avoid this by accepting extensions without signature in moneymoney or @Jacubeit solves this ;)

The same problem occures at the litecoin extension https://github.com/zafai/Litecoin-MoneyMoney

Is it also possible to use the extension with chain.io and the cloud flare firewall?

**Comment by raspulton** [comment]
Thanks! hey how do I "accept extensions without siganture" in moneymoney? I reinstalled the bitcoin extension but it still fails. Help would be very welcome!

**Comment by derausdingsda** [comment]
![Bildschirmfoto 2019-08-27 um 08 54 08](https://user-images.githubusercontent.com/7288867/63747856-610fd800-c8a8-11e9-8eae-645a214a9ca7.jpg)

**Comment by Jacubeit** [comment]
Thx guys! I uploaded v0.2 - should be signed
https://github.com/Jacubeit/Bitcoin-MoneyMoney/releases

**Comment by derausdingsda** [comment]
@Jacubeit Does it work for you? With chain.io I get:
"The server responded with the error message: Service Temporarily Unavailable. Retrying..."

As the user @mapodev wrote "chain.io not working because of firewall"

**Comment by Jacubeit** [comment]
weird - you are right - chain.io does respond with an error.
@derausdingsda can you try release v0.1 pls?

**Comment by derausdingsda** [comment]
v0.1 works for me, but it takes al lot of retrys because of bad requests by blockexplorer.

chain.io was very stable but it seems that they are using cloudflare now that produces errors.

**Comment by raspulton** [comment]
<img width="402" alt="Screenshot 2019-08-27 at 21 24 02" src="https://user-images.githubusercontent.com/54560585/63775091-3a39bc00-c911-11e9-8743-2cf63c290cb9.png">

Thank you @Jacubeit ! Version 0.1 works! Do you guys know why my Version of MoneyMoney (2.3.17) does not have the option to "Digitale Signatur von Extensions überprüfen"?

**Comment by derausdingsda** [comment]
I have the same version. Maybe you should ask moneymoney at support@moneymoney-app.com, they have good support.

**Comment by JimLoose** [comment]
unfortunately it still doesn't work because of

<img width="831" alt="grafik" src="https://user-images.githubusercontent.com/195000/64273163-0e39be80-cf41-11e9-87cd-15dcd3d9f757.png">

any solution or better provider?

**Comment by JimLoose** [comment]
it would be great if you guys can please replace chain.so with: https://www.blockchain.com/api/q

Example:
https://blockchain.info/q/addressbalance/XXXXXX?confirmations=6
make sure to add the dot starting 8 letters from the end for the right balance.

Thank you for this great extension @Jacubeit

**Comment by christophbach** [comment]
Any progress here?

@JimLoose What do you mean with "add starting dot"? I tried to use the helper function convertSatoshiToBitcoin(), but I can't get it to work.
